test/units/ansible_test/test_docker_util.py

```diff
@@ -0,0 +1,133 @@
+    def test_docker_multiple_lines(self, dc):
+        '''Test normal docker output'''
+        from ansible_test._internal.docker_util import docker_images
+
+        ret = docker_images('', 'quay.io/ansible/centos7-test-container')
+        self.assertEqual(len(ret), 3)
+
+    @patch(
+        'ansible_test._internal.docker_util.docker_command',
+        return_value=(PODMAN_OUTPUT.lstrip(), ''))
+    def test_podman_multiple(self, dc):
+        '''Test normal podman output'''
+        from ansible_test._internal.docker_util import docker_images
+
+        ret = docker_images('', 'quay.io/ansible/centos7-test-container')
+        self.assertEqual(len(ret), 2)
+
+    @patch(
+        'ansible_test._internal.docker_util.docker_command',
+        return_value=('', ''))
+    def test_no_output(self, dc):
+        '''Test empty output, which docker can emit'''
+        from ansible_test._internal.docker_util import docker_images
+
+        ret = docker_images('', 'quay.io/ansible/centos7-test-container')
+        self.assertEqual(ret, [])
+
+    def test_podman_fallback(self):
+        '''Test podman >2 && <2.2 fallback'''
+        from ansible_test._internal.docker_util import docker_images
+        from ansible_test._internal.util import SubprocessError
+
+        cmd = ['docker', 'images', 'quay.io/ansible/centos7-test-container', '--format', '{{json .}}']
+        docker_command_results = [
+            SubprocessError(cmd, status=1, stderr='function "json" not defined'),
+            (PODMAN_OUTPUT.lstrip(), ''),
+        ]
+        with patch(
+                'ansible_test._internal.docker_util.docker_command',
+                side_effect=docker_command_results) as mock:
+
+            ret = docker_images('', 'quay.io/ansible/centos7-test-container')
+            calls = [
+                call(
+                    '',
+                    ['images', 'quay.io/ansible/centos7-test-container', '--format', '{{json .}}'],
+                    capture=True,
+                    always=True),
+                call(
+                    '',
+                    ['images', 'quay.io/ansible/centos7-test-container', '--format', 'json'],
+                    capture=True,
+                    always=True),
+            ]
+            mock.assert_has_calls(calls)
+            self.assertEqual(len(ret), 2)
+
+    def test_no_such_image(self):
+        '''podman, "no such image" error'''
+        from ansible_test._internal.docker_util import docker_images
+        from ansible_test._internal.util import SubprocessError
+
+        cmd = ['docker', 'images', 'quay.io/ansible/centos7-test-container', '--format', '{{json .}}']
+        exc = SubprocessError(cmd, status=1, stderr='no such image'),
+        with patch(
+                'ansible_test._internal.docker_util.docker_command',
+                side_effect=exc) as mock:
+
+            ret = docker_images('', 'quay.io/ansible/centos7-test-container')
+            self.assertEqual(ret, [])
```

Comment:
when you opt-in to using a CM like this, make sure you only patch lines that you really need to patch. and put any assertions outside, whenever possible:
```suggestion
        assert ret == []
```